ipam/store.go

```diff
@@ -50,3 +51,47 @@ type IpamVm struct {
 	Id uint64 `sql:"AUTO_INCREMENT"`
 	//	IpamHostId sql.NullString
 }
+
+type ipamStore struct {
+	common.DbStore
+}
+
+func (ipamStore *ipamStore) addVm(stride uint, vm *Vm) error {
+	tx := ipamStore.DbStore.Db.Begin()
+
+	row := tx.Model(IpamVm{}).Where("host_id = ? AND segment_id = ?", vm.HostId, vm.SegmentId).Select("IFNULL(MAX(seq),-1)+1").Row()
```

Comment:
But you see - then you bake in your own ORM. This is why I am not huge fan of ORMs in the first place. It's a leaky abstraction always no matter what. Unless you know all the operations you can possibly do on your data - bt we do not yet know.